### PR TITLE
Standardize example analysis and formatting

### PR DESCRIPTION
**Pull request by parlough**
Sets up an `example_utils` package at the root of `/examples` which all nested examples now depend on and use as the source for their analysis. Uses `package:flutter_lints` v3 and enables a few new lints, then fixes resulting code where necessary and completes some other nearby cleanup while there.

### CODE REVIEW

**Comment by parlough** [comment]
**@parlough's blog:**

_My ability to type lessens everyday..._

File: examples/example_utils/pubspec.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+name: example_utils
```
Comment (domesticmouse) [inline examples/example_utils/pubspec.yaml:1]: I'm guessing you'll need to add this project to `.github/dependabot.yml` for the next time `flutter_lints` gets updated?

**Review by domesticmouse — APPROVED** [review]

**Comment by domesticmouse** [comment]
> **@parlough's blog:**
> 
> _My ability to type lessens everyday..._

You and me both ;-)

File: examples/example_utils/pubspec.yaml (hunk 1 above)
Comment (parlough) [reply examples/example_utils/pubspec.yaml:1]: Ah thanks for reminding me! I got distracted by fixing diagnostics that I forgot. Done now!

**Pull request merged**